OrderService/src/main/java/com/nt/order/microservice/serviceimpl/CartServiceImpl.java

```diff
@@ -0,0 +1,313 @@
+import com.nt.order.microservice.exception.UnauthorizedException;
+import com.nt.order.microservice.repository.CartRepository;
+import com.nt.order.microservice.repository.OrderRepository;
+import com.nt.order.microservice.service.CartService;
+import com.nt.order.microservice.util.Constants;
+import com.nt.order.microservice.util.Role;
+import feign.FeignException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Implementation of the CartService interface. This class handles the business logic
+ * for managing shopping cart items, such as adding items, updating quantities, and
+ * removing items from the cart.
+ */
+@Transactional
+@Service
+public class CartServiceImpl implements CartService {
+
+  private static final Logger logger = LoggerFactory.getLogger(CartServiceImpl.class);
+
+  @Autowired
+  private CartRepository cartRepository;
+
+  @Autowired
+  private OrderRepository orderRepository;
+
+  @Autowired
+  private UserFClient userFClient;
+
+  @Autowired
+  private RestaurantFClient restaurantFClient;
+
+  @Autowired
+  private FoodItemFClient foodItemFClient;
+
+  /**
+   * Adds an item to the cart for a specific user.
+   *
+   * @param cartInDTO the DTO containing the cart input data
+   * @return a response indicating the result of the operation
+   */
+  @Override
+  @Transactional
+  public CommonResponse addItemToCart(final CartInDTO cartInDTO) {
+    logger.info("Adding item to cart for userId: {}", cartInDTO.getUserId());
+
+    UserOutDTO userOutDto = fetchUser(cartInDTO.getUserId());
+    validateUserRole(userOutDto);
+    validateRestaurant(cartInDTO.getRestaurantId());
+    try {
+      validateFoodItemInRestaurant(cartInDTO);
+    } catch (Exception e) {
+      throw new ResourceNotFoundException(Constants.FoodItem_NOT_FOUND);
+    }
+
+    FoodItemOutDTO foodItemOutDTO = fetchFoodItem(cartInDTO.getFoodItemId());
+    validatePrice(cartInDTO, foodItemOutDTO);
+
+    return processCartUpdate(cartInDTO, foodItemOutDTO);
+  }
+
+  private UserOutDTO fetchUser(final Integer userId) {
+    try {
+      UserOutDTO userOutDto = userFClient.getUserProfile(userId);
+      logger.info("User fetched successfully: {}", userOutDto);
+      return userOutDto;
+    } catch (FeignException.NotFound ex) {
+      logger.error("User not found for userId: {}", userId);
+      throw new ResourceNotFoundException(Constants.USER_NOT_FOUND);
+    }
+  }
+
+  private void validateUserRole(UserOutDTO userOutDto) {
+    if (userOutDto.getRole().equals(Role.RESTAURANT_OWNER.name())) {
+      logger.warn("Restaurant owner attempting to add items to cart, userId: {}", userOutDto.getId());
```

Comment:
only one case is handled here what if the normal user is attempting to add items?